src/FuncSharp.Examples/ITry.cs

```diff
@@ -0,0 +1,235 @@
+
+            // Get method will throw an exception for unsuccessful tries. Using it is an anti-pattern.
+            // You should rather use Match to branch your code into individual cases where each case is guaranteed to work.
+            int valueOrExceptionThrown1 = iTryWithExceptionAsError.Get();
+
+            // You can also configure the exception that is thrown by mapping the error inside Get directly.
+            int valueOrExceptionThrown2 = multiplicationResult.Get(e => new Exception(e.ToString()));
+            int valueOrExceptionThrown3 = iTryWithExceptionAsError.Get(ex => new Exception("Error when multiplying", innerException: ex));
+
+            // Because ITry is a coproduct, you can check the value directly. On ITry, there are named properties for this.
+            // See IOption usages for more information about them.
+            IOption<int> successResult = multiplicationResult.Success;
+            IOption<int> successResult2 = multiplicationResult.First;
+            IOption<NetworkOperationError> errorResult = multiplicationResult.Error;
+            IOption<NetworkOperationError> errorResult2 = multiplicationResult.Second;
+        }
+
+        public void HandlingExceptionsWithCreate(decimal number, decimal divisor)
+        {
+            // Catches any exception.
+            ITry<int> divisionHandlingAllExceptions1 = Try.Create<int, Exception>(_ => DownloadNumberOverNetwork());
+
+            // Only catches a specific exception.
+            ITry<int> divisionHandlingDividingByZero = Try.Create<int, TaskCanceledException>(_ => DownloadNumberOverNetwork());
+
+            // ITry that doesn't specify the error type has a collection of exceptions by default.
+            // It is a collection because then you can aggregate multiple tries and still have the same type.
+            ITry<int, IEnumerable<Exception>> fullTypeOfVariable = divisionHandlingDividingByZero;
+        }
+
+        public void HandlingExceptionsWithCatch()
+        {
+            // Catches any exception and stores the single exception into the try.
+            // This is useful for handling individual errors, but cannot be aggregated.
+            ITry<int, Exception> divisionHandlingAllExceptions1 = Try.Catch<int, Exception>(_ => DownloadNumberOverNetwork());
+
+            // Only catches a specific exception. Notice that the error type is the specific type of exception, not a collection of exceptions.
+            ITry<int, TaskCanceledException> divisionHandlingDividingByZero = Try.Catch<int, TaskCanceledException>(_ => DownloadNumberOverNetwork());
+
+            // Catch also has an overload which allows recovering in case of exception.
+            int divisionResult = Try.Catch<int, DivideByZeroException>(
+                _ => DownloadNumberOverNetwork(),
+                exception => 42
+            );
+        }
+
+        private void ITryCreatingExamples()
```

Comment:
Actually, we could have three examples, not juts the two above. The first one would be the simple concepts of ITryCreatingExamples, GettingTryValue and UsingITryValueWithMatch. Feel free to name it ITry1.cs, ITry2.cs and ITry3.cs.